### PR TITLE
Feat/levels

### PR DESCRIPTION
**Pull request by ruihan00**
- modified movement to have incremental speed and max speed and only change dx
- modified jump to not stop movement
- added logger for debug 
- added full level in level 2

### CODE REVIEW

File: star-dash/star-dash/GameEngine/Components/BuffComponent.swift [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import Foundation
 
 class BuffComponent: Component {
-    var speedMultiplier: CGFloat = 1
+    var speedMultiplier: CGFloat = 1.5
```
Comment (Junyi00) [inline star-dash/star-dash/GameEngine/Components/BuffComponent.swift:4]: Why does BuffComponent start with a speed multiplier of 1.5?

Comment (ruihan00) [reply star-dash/star-dash/GameEngine/Components/BuffComponent.swift]: oops accident

File: star-dash/star-dash/GameEngine/Entities/EntityFactory.swift [hunk 2]
```diff
@@ -46,6 +46,7 @@ struct EntityFactory {
                         size: size)
             .withPhysics(rectangleOf: PhysicsConstants.Dimensions.monster)
                 .configureCollisionBitMask(PhysicsConstants.CollisionMask.monster)
+                .configureIsDynamic(false)
```
Comment (HoJunHao2000) [inline star-dash/star-dash/GameEngine/Entities/EntityFactory.swift:49]: FYI. In my feat/monsters branch i set it as dynamic to allow it move about and collide. But you can leave this in for now.

**Review by HoJunHao2000 — COMMENTED** [review]
Is this an intended effect? When i tap and hold the end of the joystick, player moves super slowly. Only when I drag the joystick the player moves about normally.

https://github.com/cs3217-2324/StarDash/assets/84698566/095e26c4-7693-4b25-9a47-88ca6c650f7a

**Review by HoJunHao2000 — APPROVED** [review]
lgtm

**Pull request merged**